lib/auth/access_request_test.go

```diff
@@ -1518,3 +1519,122 @@ func TestUpdateAccessRequestWithAdditionalReviewers(t *testing.T) {
+
+	t.Run("CreateAccessRequest, request a specific start time", func(t *testing.T) {
+		// create the access request object
+		req, err := services.NewAccessRequest(requesterUserName, "admins")
+		require.NoError(t, err)
+		req.SetMaxDuration(maxDuration)
+
+		// invalid, greater than constants.MaxAssumeStartDuration
+		req.SetAssumeStartTime(invalidMaxedAssumeStartTime)
+		_, err = requesterClient.CreateAccessRequestV2(ctx, req)
+		require.True(t, trace.IsBadParameter(err), "expected bad parameter, got %v", err)
+		require.Contains(t, err.Error(), "assume start time is too far in the future")
+
+		// invalid, after access expiry time
+		req.SetAssumeStartTime(invalidExpiredAssumeStartTime)
+		_, err = requesterClient.CreateAccessRequestV2(ctx, req)
+		require.True(t, trace.IsBadParameter(err), "expected bad parameter, got %v", err)
+		require.Contains(t, err.Error(), "assume start time cannot equal or exceed access expiry time")
+
+		// valid start time
+		req.SetAssumeStartTime(validStartTime)
+		createdReq, err = requesterClient.CreateAccessRequestV2(ctx, req)
+		require.NoError(t, err)
+		require.Equal(t, validStartTime, *createdReq.GetAssumeStartTime())
+	})
+
+	var changedStartTime time.Time
+	t.Run("SubmitAccessReview, initial change requested start time", func(t *testing.T) {
+		review := types.AccessReviewSubmission{
+			RequestID: createdReq.GetName(),
+			Review: types.AccessReview{
+				Author:        "admin",
+				ProposedState: types.RequestState_APPROVED,
+			},
+		}
+
+		// invalid, greater than constants.MaxAssumeStartDuration
+		review.Review.AssumeStartTime = &invalidMaxedAssumeStartTime
+		_, err := testPack.tlsServer.AuthServer.AuthServer.SubmitAccessReview(ctx, review)
+		require.True(t, trace.IsBadParameter(err), "expected bad parameter, got %v", err)
+		require.Contains(t, err.Error(), "assume start time is too far in the future")
+
+		// invalid, after access expiry time
+		review.Review.AssumeStartTime = &invalidExpiredAssumeStartTime
+		_, err = testPack.tlsServer.AuthServer.AuthServer.SubmitAccessReview(ctx, review)
+		require.True(t, trace.IsBadParameter(err), "expected bad parameter, got %v", err)
+		require.Contains(t, err.Error(), "assume start time cannot equal or exceed access expiry time")
+
+		// valid, changed start time
+		changedStartTime = validStartTime.Add(-day * 2)
+		review.Review.AssumeStartTime = &changedStartTime
+		resp, err := testPack.tlsServer.AuthServer.AuthServer.SubmitAccessReview(ctx, review)
+		require.NoError(t, err)
+		require.Equal(t, changedStartTime, *resp.GetAssumeStartTime())
+	})
+
+	t.Run("SetAccessRequestState, subsequent change changed start time", func(t *testing.T) {
+		// double check current assume start time was from previous results.
+		resp, err := testPack.tlsServer.AuthServer.AuthServer.GetAccessRequests(ctx, types.AccessRequestFilter{})
+		require.NoError(t, err)
+		require.Len(t, resp, 1)
+		require.Equal(t, changedStartTime, *resp[0].GetAssumeStartTime())
```

Comment:
I'm not a big fan of sub tests that depend on the result of previous sub tests.

It's possible to run an individual subtest with `go test -run ...`, but it seems like that would fail here because this test depends on the value of `changedStartTime` from a previous test.

Let's aim to have each test independent of the others.